### PR TITLE
Add single frame encoder for type-gif

### PR DESCRIPTION
**Pull request by jeffbseeking**
# What's Changing and Why
There wasn't a encoder for image/gif so any attempt to create a buffer with mimetype image/gif would fail with the error Unsupported MIME type: image/gif

## What else might be affected
Not sure how things hook up with the getBuffer method so could use some guidance here.  All the other encoder functions in the other types seemed synchronous.

## Tasks

- [X] Add tests
- [ ] Update Documentation
- [N/A] Update `jimp.d.ts`
- [X] Add [SemVer](https://semver.org/) Label

<details>
  <summary>📦 Published PR as canary version: <code>0.12.2-canary.899.952.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```sh
  npm install @jimp/cli@0.12.2-canary.899.952.0
  npm install @jimp/core@0.12.2-canary.899.952.0
  npm install @jimp/custom@0.12.2-canary.899.952.0
  npm install jimp@0.12.2-canary.899.952.0
  npm install @jimp/plugin-blit@0.12.2-canary.899.952.0
  npm install @jimp/plugin-blur@0.12.2-canary.899.952.0
  npm install @jimp/plugin-circle@0.12.2-canary.899.952.0
  npm install @jimp/plugin-color@0.12.2-canary.899.952.0
  npm install @jimp/plugin-contain@0.12.2-canary.899.952.0
  npm install @jimp/plugin-cover@0.12.2-canary.899.952.0
  npm install @jimp/plugin-crop@0.12.2-canary.899.952.0
  npm install @jimp/plugin-displace@0.12.2-canary.899.952.0
  npm install @jimp/plugin-dither@0.12.2-canary.899.952.0
  npm install @jimp/plugin-fisheye@0.12.2-canary.899.952.0
  npm install @jimp/plugin-flip@0.12.2-canary.899.952.0
  npm install @jimp/plugin-gaussian@0.12.2-canary.899.952.0
  npm install @jimp/plugin-invert@0.12.2-canary.899.952.0
  npm install @jimp/plugin-mask@0.12.2-canary.899.952.0
  npm install @jimp/plugin-normalize@0.12.2-canary.899.952.0
  npm install @jimp/plugin-print@0.12.2-canary.899.952.0
  npm install @jimp/plugin-resize@0.12.2-canary.899.952.0
  npm install @jimp/plugin-rotate@0.12.2-canary.899.952.0
  npm install @jimp/plugin-scale@0.12.2-canary.899.952.0
  npm install @jimp/plugin-shadow@0.12.2-canary.899.952.0
  npm install @jimp/plugin-threshold@0.12.2-canary.899.952.0
  npm install @jimp/plugins@0.12.2-canary.899.952.0
  npm install @jimp/test-utils@0.12.2-canary.899.952.0
  npm install @jimp/bmp@0.12.2-canary.899.952.0
  npm install @jimp/gif@0.12.2-canary.899.952.0
  npm install @jimp/jpeg@0.12.2-canary.899.952.0
  npm install @jimp/png@0.12.2-canary.899.952.0
  npm install @jimp/tiff@0.12.2-canary.899.952.0
  npm install @jimp/types@0.12.2-canary.899.952.0
  npm install @jimp/utils@0.12.2-canary.899.952.0
  # or 
  yarn add @jimp/cli@0.12.2-canary.899.952.0
  yarn add @jimp/core@0.12.2-canary.899.952.0
  yarn add @jimp/custom@0.12.2-canary.899.952.0
  yarn add jimp@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-blit@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-blur@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-circle@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-color@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-contain@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-cover@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-crop@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-displace@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-dither@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-fisheye@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-flip@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-gaussian@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-invert@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-mask@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-normalize@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-print@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-resize@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-rotate@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-scale@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-shadow@0.12.2-canary.899.952.0
  yarn add @jimp/plugin-threshold@0.12.2-canary.899.952.0
  yarn add @jimp/plugins@0.12.2-canary.899.952.0
  yarn add @jimp/test-utils@0.12.2-canary.899.952.0
  yarn add @jimp/bmp@0.12.2-canary.899.952.0
  yarn add @jimp/gif@0.12.2-canary.899.952.0
  yarn add @jimp/jpeg@0.12.2-canary.899.952.0
  yarn add @jimp/png@0.12.2-canary.899.952.0
  yarn add @jimp/tiff@0.12.2-canary.899.952.0
  yarn add @jimp/types@0.12.2-canary.899.952.0
  yarn add @jimp/utils@0.12.2-canary.899.952.0
  ```
</details>

### CODE REVIEW

**Comment by hipstersmoothie** [comment]
This is looking good. get the lint passing and I'll merge 🚀

**Comment by jeffbseeking** [comment]
Fixed linting @hipstersmoothie

File: packages/type-gif/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@jimp/gif",
-  "version": "0.12.1",
+  "version": "0.13.0",
```
Comment (hipstersmoothie) [inline packages/type-gif/package.json:3]:
```suggestion
  "version": "0.12.1",
```

**Comment by hipstersmoothie** [comment]
The version for this project is automated through GitHub labels using https://github.com/intuit/auto. Thanks for thinking about it though!

**Comment by hipstersmoothie** [comment]
:rocket: PR was released in `v0.13.0` :rocket:

**Pull request merged**